### ISSUE TITLE
I have something non-atomic and I don't want it's children generating tokens

### ISSUE THREAD

**Issue by rsnakard-r7** [pest-parser/pest#439]
I'm parsing quoted strings and when I receive a quoted string I want to receive it as one big chunk, I don't need to see the tokens that comprise it. It's very frustrating.

```
quote = { "\"" }
segment = @{ ASCII_ALPHA+ }
quoted_string = { quote ~ segment+ ~ quote }
string = { (segment | quoted_string)+ }
```

If I'm parsing `The quick brown "fox jumps over the" lazy dog` I would like to receive the tokens
```
Rule::string - "The quick brown "fox jumps over the" lazy dog"
Rule::segment - "The"
Rule::segment - "quick"
Rule::quoted_string - "fox jumps over the"
Rule::segment - "lazy"
Rule::segment - "dog"
```
This will make processing much easier as I can match against `Rule::segment` without worrying if it's inner of the `quoted_string` or not. I realize I have the option of adding a wrapper rule
```
unquoted_string = { segment }
```
This will allow me to match against `quoted_string` and `unquoted_string` without fear but it does leave my iterator clogged up with lots of segments that I just don't need.

There is currently no way to mark `quoted_string` as "not atomic but don't generate tokens for my children" and there is currently no way to mark `segment` as "atomic but don't generate tokens for me". One of these two features is necessary for my use case

**Comment by mdg** [comment]
I wonder if you could clarify  why you don't want it to be atomic?

**Comment by CAD97** [comment]
Yes, this is one of the main pain points we want to address with the next major version of pest. (Unfortunately, it's kind of stalled currently.) The multiple implications of each of the modifiers (and only being able to state one) doesn't least to composable control over behavior.

The current advice would be to just recognize the string rule when consuming the parse tree and not recursing into it.
